libr/core/cmd_anal.c

```diff
@@ -6072,6 +6073,105 @@ static void __anal_esil_function(RCore *core, ut64 addr) {
 	r_anal_esil_free (core->anal->esil);
 }
 
+static void print_esil_dfg_as_commands(RCore *core, RAnalEsilDFG *dfg) {
+	RListIter *iter, *ator;
+	RGraphNode *node, *edon;
+	RStrBuf *sb = r_strbuf_new ("");
+	r_cons_println ("ag-");
+	r_list_foreach (r_graph_get_nodes (dfg->flow), iter, node) {
+		const RAnalEsilDFGNode *enode = (RAnalEsilDFGNode *)node->data;
+		char *esc_str = r_str_escape (r_strbuf_get (enode->content));
+		r_strbuf_set (sb, esc_str);
+		if (enode->type == R_ANAL_ESIL_DFG_BLOCK_GENERATIVE) {
+			r_strbuf_prepend (sb, "generative:");
+		}
+		char *b64_buf = r_base64_encode_dyn (r_strbuf_get (sb), sb->len);
+		if (!b64_buf) {
+			r_strbuf_free (sb);
+			free (b64_buf);
+			free (esc_str);
+			return;
+		}
+
+		r_cons_printf ("agn %d base64:%s\n", enode->idx, b64_buf);
+		free (b64_buf);
+		free (esc_str);
+	}
+	r_strbuf_free (sb);
+
+	r_list_foreach (r_graph_get_nodes (dfg->flow), iter, node) {
+		const RAnalEsilDFGNode *enode = (RAnalEsilDFGNode *)node->data;
+		r_list_foreach (r_graph_get_neighbours (dfg->flow, node), ator, edon) {
+			const RAnalEsilDFGNode *edone = (RAnalEsilDFGNode *)edon->data;
+			r_cons_printf ("age %d %d\n", enode->idx, edone->idx);
+		}
+	}
+}
+
+static void cmd_aeg(RCore *core, int argc, char *argv[]) {
+	if (!argc) {
+		r_core_cmd0 (core, ".aeg*;agg");
+		return;
+	}
+	if (argc == 1) {	//aeg
+		RAnalEsilDFG *dfg = r_anal_esil_dfg_expr (core->anal, NULL, argv[0]);
+		r_return_if_fail (dfg);
+		print_esil_dfg_as_commands (core, dfg);
+		r_anal_esil_dfg_free (dfg);
+		return;
+	}
+	if (argv[0][0] == '*') {	//aeg*
+		//does this memleak?
+		RAnalOp *aop = r_core_anal_op (core, core->offset, R_ANAL_OP_MASK_ESIL);
+		r_return_if_fail (aop);
+		const char *esilstr = r_strbuf_get (&aop->esil);
+		if (R_STR_ISNOTEMPTY (esilstr)) {
+			RAnalEsilDFG *dfg = r_anal_esil_dfg_expr (core->anal, NULL, esilstr);
+			r_return_if_fail (dfg);
```

Comment:
uhm is this possible to happen or to catch possible bugs in the code?

Reply:
it should only return NULL, if R_NEW0 fails